.github/workflows/bikes.yml

```diff
@@ -29,32 +28,70 @@ jobs:
         creds: '${{ secrets.AZURE_CREDENTIALS }}'
         cluster-name: ${{ secrets.CLUSTER_NAME }}
         resource-group: ${{ secrets.RESOURCE_GROUP }}
-                
+
+    - name: Create valid child namespace name
+      id: generate-child-space-name
+      run: |
+        branch=${GITHUB_HEAD_REF::63}        
+        branch=$(echo ${branch//[!0-9a-zA-Z]/-} | tr '[:upper:]' '[:lower:]' | sed -e 's/^-/z/' -e 's/-$/z/')
+        echo ::set-output name=result::$branch         
+    
     - name: Create child namespaces and add Dev Spaces labels
       run:  |
-        if [ -n "$(kubectl get namespace $GITHUB_HEAD_REF --ignore-not-found -o name)" ]; then exit 0; fi
-        kubectl create namespace $GITHUB_HEAD_REF
-        kubectl label namespace $GITHUB_HEAD_REF azds.io/parent-space=${{ secrets.MASTER_SPACE }} 
-        kubectl label namespace $GITHUB_HEAD_REF azds.io/space=true
-        kubectl label namespace $GITHUB_HEAD_REF azds.io/pull-request-space=true
+        if [ -n "$(kubectl get namespace ${{steps.generate-child-space-name.outputs.result}} --ignore-not-found -o name)" ]; then exit 0; fi
+        kubectl create namespace ${{steps.generate-child-space-name.outputs.result}}
+        kubectl label namespace ${{steps.generate-child-space-name.outputs.result}} azds.io/parent-space=${{ secrets.MASTER_SPACE }} 
+        kubectl label namespace ${{steps.generate-child-space-name.outputs.result}} azds.io/space=true
+        kubectl label namespace ${{steps.generate-child-space-name.outputs.result}} azds.io/pull-request-space=true
         
     - uses: azure/k8s-actions/k8s-create-secret@master
       with:
-        namespace: ${{ github.head_ref }}
+        namespace: ${{ steps.generate-child-space-name.outputs.result }}
         container-registry-url: ${{ secrets.CONTAINER_REGISTRY }}
         container-registry-username: ${{ secrets.REGISTRY_USERNAME }}
         container-registry-password: ${{ secrets.REGISTRY_PASSWORD }}
         secret-name: ${{ secrets.IMAGE_PULL_SECRET }}
-        
-    - name: Helm Install PR
+
+    - name: Create valid release-name
+      id: generate-release-name
       run: |
-        helm init --upgrade --wait
-        helm upgrade --install --namespace $GITHUB_HEAD_REF bikesharing-$GITHUB_HEAD_REF samples/BikeSharingApp/Bikes/charts/bikes \
+        release=${bikesharing-$GITHUB_HEAD_REF}
+        release=${release::53}
+        release=$(echo ${release//[!0-9a-zA-Z]/-} | tr '[:upper:]' '[:lower:]' | sed -e 's/^-/z/' -e 's/-$/z/')
+        echo ::set-output name=result::$release
+
+    - name: Install Helm for client
+      id: install-helm-client
+      run: |
+        clientversion=$(echo $(helm version --client --short) | sed -n 's/.*Client: \(.*\)+.*/\1/p')
```

Comment:
Not sure why you need `echo` here, Does this not work?

`clientversion=$(helm version --client --short | sed -n 's/.*Client: \(.*\)+.*/\1/p')`

Also you don't need sed here. Simplify to:

`clientversion=$(helm version --client --short | cut -d' ' -f2)`

Reply:
clientversion=$(helm version --client --short | cut -d' ' -f2) ... does not work because this command does not give us the version number which we need in the subsequent steps.
I modified it  (and similarly for serverversion) to:
clientversion=$(helm version --client --short | cut -d' ' -f2 | cut -d'+' -f1)